### DISCUSSION TITLE
Dependency Injection Annotation Strategy

### DISCUSSION THREAD

**Discussion by ghost** [pointfreeco/swift-composable-architecture#1432] [Q&A] [answered]
Hi 

I am working on something using TCA and wanted to use [Resolver](https://github.com/hmlongco/Resolver)  Dependency Injection / Service Locator framework

Please see screenshot, 
Problem: How can I pass data from A View to B View's State by using service locator.
Reference to my problem: https://github.com/hmlongco/Resolver/blob/master/Documentation/Arguments.md
Thanks.

<img width="689" alt="Screenshot 2022-10-01 at 20 41 53" src="https://user-images.githubusercontent.com/98336263/193425782-6ca34bdb-e33d-460e-9849-0dd7db9b92f7.png">
<img width="691" alt="Screenshot 2022-10-01 at 20 43 09" src="https://user-images.githubusercontent.com/98336263/193425790-f4f14554-9573-4511-92ff-b0ddd2dc7934.png">

**Comment by mbrandonw** [comment, marked as answer]
Can you explain more about what you are trying to accomplish?

In general I don't think it's necessary to inject stores. The store a view holds onto is typically derived from its parent view, and so it is dynamic and not something that can be statically known at the root of the app and made available to any view.

Where dependency injection does make more sense is when it comes to dependencies that encapsulate side effects. In TCA today you pass those kinds of dependencies explicitly via the environment, but soon that will be done via a new dependency management system. See [the announcement](https://github.com/pointfreeco/swift-composable-architecture/discussions/1282) for more information.

**Reply by ghost** [reply]
Just watched this https://www.pointfree.co/episodes/ep205-reducer-protocol-dependencies-part-1
This not just answers what I wanna do, also showed me a great perspective for DI. 
Thanks
